beacon_node/network/src/sync/range_sync/chain.rs

```diff
@@ -744,6 +763,23 @@ impl<T: BeaconChainTypes> SyncingChain<T> {
         self.state = ChainSyncingState::Stopped;
     }
 
+    pub fn execution_stalled(&mut self) {
+        self.state = ChainSyncingState::ExecutionStalled;
+    }
+
+    pub fn execution_resumed(
+        &mut self,
+        network: &mut SyncNetworkContext<T::EthSpec>,
+    ) -> ProcessingResult {
+        if let ChainSyncingState::ExecutionStalled = self.state {
+            self.state = ChainSyncingState::Syncing;
+            return self.request_batches(network);
+        }
+        Err(RemoveChain::WrongBatchState(
+            "Invalid batch state".to_string(),
+        ))
+    }
```

Comment:
I didn't find references to the `execution_stalled` function. I would expect this to be called fr other chains because the failure is identified at the chain level but rn not propagated up. So, if one chain gets the error, it stalls, and on resume, all chains are resumed, they find themselves in a wrong state and ask to be removed. I think this is hard to notice is tests / masked since sync is self-resilient

Reply:
Nice catch. My thinking was each chain will get stalled once they get an offline execution error on each of the chains. But didn't consider that they might go into an invalid state if it we resume before it goes into waiting.